opsmngr/checkpoints.go

```diff
@@ -0,0 +1,74 @@
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	checkpoints = "groups/%s/clusters/%s/checkpoints"
+)
+
+// CheckpointsService is an interface for interfacing with the Checkpoint
+// endpoints of the MongoDB Atlas API.
+type CheckpointsService interface {
+	List(context.Context, string, string, *atlas.ListOptions) (*atlas.Checkpoints, *atlas.Response, error)
+	Get(context.Context, string, string, string) (*atlas.Checkpoint, *atlas.Response, error)
+}
+
+// CheckpointsServiceOp handles communication with the checkpoint related methods of the
+// MongoDB Atlas API
+type CheckpointsServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+var _ CheckpointsService = &CheckpointsServiceOp{}
+
+func (s CheckpointsServiceOp) List(ctx context.Context, groupID, clusterName string, listOptions *atlas.ListOptions) (*atlas.Checkpoints, *atlas.Response, error) {
```

Comment:
can we make s a pointer? 
```suggestion
func (s *CheckpointsServiceOp) List(ctx context.Context, groupID, clusterName string, listOptions *atlas.ListOptions) (*atlas.Checkpoints, *atlas.Response, error) {
```

On using or not using pointers here is a trick question in go and I haven't found a good answer yet. What most people recommend is that for small structs no pointer is ok, but structs that grow a pointer may handle memory better. So far my request here comes from consistency with other services where we use pointers